src/server/services/announcement.service.ts

```diff
@@ -27,6 +27,7 @@ export const getLatestAnnouncement = async <TSelect extends Prisma.AnnouncementS
   });
 };
 
+export type GetAnnouncement = Awaited<ReturnType<typeof getAnnouncements>>[number];
 export const getAnnouncements = async ({ dismissed, ids }: GetAnnouncementsInput) => {
   const now = new Date();
   return dbRead.announcement.findMany({
```

Comment:
I think we can do the casting of the metadata in here
```ts
  const announcements = await dbRead.announcement.findMany({
    // ...
  });

  return announcements.map(({ metadata, ...item }) => ({ ...item, metadata: metadata as AnnouncementMetaSchema }))
```

Reply:
Good thinking 👍